File: src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.CSharp.Codefix.Testing.xUnit/CodefixVerifier.cs
using Microsoft.CodeAnalysis.CodeFixes;
using Microsoft.CodeAnalysis.Diagnostics;

namespace Microsoft.CodeAnalysis.CSharp.Testing.xUnit
{
    public static class CodefixVerifier
    {
        public static CodefixVerifier<TAnalyzer, TCodefix> Create<TAnalyzer, TCodefix>()
            where TAnalyzer : DiagnosticAnalyzer, new()
            where TCodefix : CodeFixProvider, new()
        {
            return new CodefixVerifier<TAnalyzer, TCodefix>();
        }
    }
}

File: src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.VisualBasic.Codefix.Testing.XUnit/CodeFixVerifier.vb
Imports Microsoft.CodeAnalysis.CodeFixes
Imports Microsoft.CodeAnalysis.Diagnostics

Namespace Microsoft.CodeAnalysis.VisualBasic.Testing.XUnit
    Module CodeFixVerifier
        Public Function Create(Of TAnalyzer As {DiagnosticAnalyzer, New}, TCodefix As {CodeFixProvider, New})() As CodefixVerifier(Of TAnalyzer, TCodefix)
            Return New CodefixVerifier(Of TAnalyzer, TCodefix)
        End Function
    End Module
End Namespace
